help.go

```diff
@@ -42,9 +42,13 @@ func (e *Executor) printTasks(listAll bool) {
 	e.Logger.Outf(logger.Default, "task: Available tasks for this project:")
 
 	// Format in tab-separated columns with a tab stop of 8.
-	w := tabwriter.NewWriter(e.Stdout, 0, 8, 0, '\t', 0)
+	w := tabwriter.NewWriter(e.Stdout, 0, 8, 6, ' ', 0)
 	for _, task := range tasks {
-		fmt.Fprintf(w, "* %s: \t%s\n", task.Name(), task.Desc)
+		fmt.Fprintf(w, "* ")
+		e.Logger.FOutf(w, logger.Green, task.Name())
+		fmt.Fprintf(w, ": \t")
+		e.Logger.FOutf(w, logger.Cyan, task.Desc)
+		fmt.Fprintf(w, "\n")
```

Comment:
For consistency you could use `e.Logger.FOutf` for all calls, but using `logger.Default` for the white texts.

Reply:
I did give this a go, but it resulted in this weird spacing issue and I couldn't figure out what was going on (see red mark in screenshot below).

```diff
diff --git a/help.go b/help.go
index 3457683..4914347 100644
--- a/help.go
+++ b/help.go
@@ -44,11 +44,11 @@ func (e *Executor) printTasks(listAll bool) {
        // Format in tab-separated columns with a tab stop of 8.
        w := tabwriter.NewWriter(e.Stdout, 0, 8, 6, ' ', 0)
        for _, task := range tasks {
-               fmt.Fprintf(w, "* ")
+               e.Logger.FOutf(w, logger.Default, "* ")
                e.Logger.FOutf(w, logger.Green, task.Name())
-               fmt.Fprintf(w, ": \t")
+               e.Logger.FOutf(w, logger.Default, ": \t")
                e.Logger.FOutf(w, logger.Cyan, task.Desc)
-               fmt.Fprintf(w, "\n")
+               e.Logger.FOutf(w, logger.Default, "\n")
        }
        w.Flush()
 }
```

![image](https://user-images.githubusercontent.com/9294862/193167244-b8fd8fd0-0cf0-464a-9ea8-e16818704393.png)

Reply:
I suppose it's fine to keep as is then...

Reply:
I agree it would be nicer to be consistent. I'll take another look at this tomorrow and see if I can figure out what's going on.

Reply:
I had a bit of time to look at this today and worked out what's going on. This first thing to note is that `text/tabwriter` doesn't care about escape sequences. This means that when you add color escape sequences, you can get some strange behaviour.

To make debugging easier, I replaced the tab writer with a regular `bytes.buffer` and printed the output. This is what we were getting (I've reformatted this slightly so its easier to tell what's going on):

```
"       \x1b[33m* \x1b[0m\x1b[32m <task_1> \x1b[0m\x1b[0m: \t <task 1 desc>"
"\x1b[0m\x1b[33m* \x1b[0m\x1b[32m <task_2> \x1b[0m\x1b[0m: \t <task 2 desc>"
...
"\x1b[0m\x1b[33m* \x1b[0m\x1b[32m <task_n> \x1b[0m\x1b[0m: \t <task n desc>"
"\x1b[0m"
```

As you can see, because each color print function _wraps_ the text sent to the writer, when you send a `\n` to it, the color reset escape sequence will be printed on a new line. This means that the tabwriter package will see the next line (and all subsequent lines) as longer than the first line and this leads to oddities with the columns.

As there is no way to change this behaviour in `text/tabwriter` (I've tried using `tabwriter.StripEscape` to no effect). I propose that we consistently use `e.Logger.FOutf` for all output except the final newline which will use `fmt.Fprint`. This will result in the following:

```
"\x1b[33m* \x1b[0m\x1b[32m <task_1>           \x1b[0m\x1b[0m: \t <task_1_desc> \x1b[0m"
"\x1b[33m* \x1b[0m\x1b[32m <task_2>           \x1b[0m\x1b[0m: \t <task_2_desc> \x1b[0m"
...
"\x1b[33m* \x1b[0m\x1b[32m <task_n>           \x1b[0m\x1b[0m: \t <task_n_desc> \x1b[0m"
```

Because the escape sequences are consistent on every line, `text/tabwriter` will now deal with it correctly.